### PR TITLE
feat: Add `install` support for Transformer plugins

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

This PR allows for transformer plugins to be downloaded by the `cloudquery install` command

### CODE REVIEW

File: cli/internal/auth/token.go [hunk 1]
```diff
@@ -12,6 +12,15 @@ func tokenNeeded(registry specs.Registry, path string) bool {
 	return registry == specs.RegistryCloudQuery || (registry == specs.RegistryDocker && strings.HasPrefix(path, "docker.cloudquery.io"))
 }
 
+func transformerNeedsToken(sources []*specs.Transformer) bool {
```
Comment (disq) [inline cli/internal/auth/token.go:15]:
```suggestion
func transformerNeedsToken(transformers []*specs.Transformer) bool {
```

File: cli/internal/auth/token.go [hunk 2]
```diff
@@ -12,6 +12,15 @@ func tokenNeeded(registry specs.Registry, path string) bool {
...
+	for _, source := range sources {
```
Comment (disq) [inline cli/internal/auth/token.go:16]:
```suggestion
	for _, transformer := range transformers {
```

File: cli/internal/auth/token.go [hunk 3]
```diff
@@ -12,6 +12,15 @@ func tokenNeeded(registry specs.Registry, path string) bool {
...
+		if tokenNeeded(source.Registry, source.Path) {
```
Comment (disq) [inline cli/internal/auth/token.go:17]:
```suggestion
		if tokenNeeded(transformer.Registry, transformer.Path) {
```

File: cli/internal/auth/token.go (hunk 1 above)
Comment (bbernays) [reply cli/internal/auth/token.go]: Fixed in [a5c61c1](https://github.com/cloudquery/cloudquery/pull/19166/commits/a5c61c113b44cf4ca767a164e824a9f0e06600a7)

File: cli/internal/auth/token.go (hunk 3 above)
Comment (bbernays) [reply cli/internal/auth/token.go]: [a5c61c1](https://github.com/cloudquery/cloudquery/pull/19166/commits/a5c61c113b44cf4ca767a164e824a9f0e06600a7)

File: cli/internal/auth/token.go (hunk 2 above)
Comment (bbernays) [reply cli/internal/auth/token.go]: [a5c61c1](https://github.com/cloudquery/cloudquery/pull/19166/commits/a5c61c113b44cf4ca767a164e824a9f0e06600a7)

**Review by disq — APPROVED** [review]

**Pull request merged**